Mathlib/RingTheory/ClassGroup.lean

```diff
@@ -137,11 +137,11 @@ theorem ClassGroup.mk_eq_mk_of_coe_ideal {I J : (FractionalIdeal R⁰ <| Fractio
       sec_snd_ne_zero (R := R) le_rfl (x : FractionRing R), hJ⟩
   · rintro ⟨x, y, hx, hy, h⟩
     constructor
-    rw [mul_comm, ← Units.eq_iff, Units.val_mul, coe_toPrincipalIdeal]
-    convert
-      (mk'_mul_coeIdeal_eq_coeIdeal (FractionRing R) <| mem_nonZeroDivisors_of_ne_zero hy).2 h
-    apply (Ne.isUnit _).unit_spec
-    rwa [Ne, mk'_eq_zero_iff_eq_zero]
+    · rw [mul_comm, ← Units.eq_iff, Units.val_mul, coe_toPrincipalIdeal]
```

Comment:
This one looks pretty bad, since there are two goals, the second of which is data and is constructed while proving the first goal, so in the end there is only one goal. Maybe change it to
```lean
    have : IsUnit (mk' (FractionRing R) x ⟨y, mem_nonZeroDivisors_of_ne_zero hy⟩) := by
      simpa only [isUnit_iff_ne_zero, ne_eq, mk'_eq_zero_iff_eq_zero] using hx
    refine ⟨this.unit, ?_⟩
    rw [mul_comm, ← Units.eq_iff, Units.val_mul, coe_toPrincipalIdeal]
    convert
      (mk'_mul_coeIdeal_eq_coeIdeal (FractionRing R) <| mem_nonZeroDivisors_of_ne_zero hy).2 h
```
which is less magical, but not really longer.

Reply:
Sébastien, thank you for your comment!  I used your version.

While preparing the PR, I tried to be as minimal as possible in the changes, and I (almost) always only changed `·` and indents.